metadata-ingestion/src/datahub/ingestion/source/sql/mssql/source.py

```diff
@@ -268,6 +636,73 @@ def _get_columns(
                 column["comment"] = description
         return columns
 
+    def _process_table(
+        self,
+        dataset_name: str,
+        inspector: Inspector,
+        schema: str,
+        table: str,
+        sql_config: SQLCommonConfig,
+        data_reader: Optional[DataReader],
+    ) -> Iterable[Union[SqlWorkUnit, MetadataWorkUnit]]:
+        yield from super()._process_table(
+            dataset_name=dataset_name,
+            inspector=inspector,
+            schema=schema,
+            table=table,
+            sql_config=sql_config,
+            data_reader=data_reader,
+        )
+
+        dataset_urn = make_dataset_urn_with_platform_instance(
+            self.platform,
+            dataset_name,
+            self.config.platform_instance,
+            self.config.env,
+        )
+
+        db_name = self.get_db_name(inspector)
+        _key = f"{db_name}.{schema}.{table}"
+
+        upstreams = self.get_ucs(inspector=inspector, key=_key)
+        if upstreams:
+            yield MetadataChangeProposalWrapper(
+                entityUrn=dataset_urn,
+                aspect=UpstreamLineage(upstreams=upstreams),
+            ).as_workunit()
+
+    def _process_view(
+        self,
+        dataset_name: str,
+        inspector: Inspector,
+        schema: str,
+        view: str,
+        sql_config: SQLCommonConfig,
+    ) -> Iterable[Union[SqlWorkUnit, MetadataWorkUnit]]:
+        yield from super()._process_view(
+            dataset_name=dataset_name,
+            inspector=inspector,
+            schema=schema,
+            view=view,
+            sql_config=sql_config,
+        )
+
+        dataset_urn = make_dataset_urn_with_platform_instance(
+            self.platform,
+            dataset_name,
+            self.config.platform_instance,
+            self.config.env,
+        )
+        db_name = self.get_db_name(inspector)
+        _key = f"{db_name}.{schema}.{view}"
+
+        upstreams = self.get_ucs(inspector=inspector, key=_key)
+        if upstreams:
+            yield MetadataChangeProposalWrapper(
+                entityUrn=dataset_urn,
+                aspect=UpstreamLineage(upstreams=upstreams),
```

Comment:
we also try to generate view lineage using sql parsing. That is capable of generating column lineage as well

This will likely overwrite that stuff, which doesn't feel right